src/main/java/com/fasterxml/jackson/databind/util/concurrentlinkedhashmap/Weighers.java

```diff
@@ -0,0 +1,282 @@
+     * key-value pairs in the map.
+     * <p>
+     * A value with a weight of <tt>0</tt> will be rejected by the map. If a value
+     * with this weight can occur then the caller should eagerly evaluate the
+     * value and treat it as a removal operation. Alternatively, a custom weigher
+     * may be specified on the map to assign an empty value a positive weight.
+     *
+     * @return A weigher where each element takes one unit of capacity.
+     */
+    @SuppressWarnings({"cast", "unchecked"})
+    public static <E> Weigher<? super List<E>> list() {
+        return (Weigher<List<E>>) (Weigher<?>) ListWeigher.INSTANCE;
+    }
+
+    /**
+     * A weigher where the value is a {@link Set} and its weight is the number
+     * of elements. A map bounded with this weigher will evict when the total
+     * number of elements exceeds the capacity rather than the number of
+     * key-value pairs in the map.
+     * <p>
+     * A value with a weight of <tt>0</tt> will be rejected by the map. If a value
+     * with this weight can occur then the caller should eagerly evaluate the
+     * value and treat it as a removal operation. Alternatively, a custom weigher
+     * may be specified on the map to assign an empty value a positive weight.
+     *
+     * @return A weigher where each element takes one unit of capacity.
+     */
+    @SuppressWarnings({"cast", "unchecked"})
+    public static <E> Weigher<? super Set<E>> set() {
+        return (Weigher<Set<E>>) (Weigher<?>) SetWeigher.INSTANCE;
+    }
+
+    /**
+     * A weigher where the value is a {@link Map} and its weight is the number of
+     * entries. A map bounded with this weigher will evict when the total number of
+     * entries across all values exceeds the capacity rather than the number of
+     * key-value pairs in the map.
+     * <p>
+     * A value with a weight of <tt>0</tt> will be rejected by the map. If a value
+     * with this weight can occur then the caller should eagerly evaluate the
+     * value and treat it as a removal operation. Alternatively, a custom weigher
+     * may be specified on the map to assign an empty value a positive weight.
+     *
+     * @return A weigher where each entry takes one unit of capacity.
+     */
+    @SuppressWarnings({"cast", "unchecked"})
+    public static <A, B> Weigher<? super Map<A, B>> map() {
+        return (Weigher<Map<A, B>>) (Weigher<?>) MapWeigher.INSTANCE;
+    }
+
+    static final class EntryWeigherView<K, V> implements EntryWeigher<K, V>, Serializable {
+        static final long serialVersionUID = 1;
+        final Weigher<? super V> weigher;
+
+        EntryWeigherView(Weigher<? super V> weigher) {
+            checkNotNull(weigher);
+            this.weigher = weigher;
+        }
+
+        @Override
+        public int weightOf(K key, V value) {
+            return weigher.weightOf(value);
+        }
+    }
+
+    enum SingletonEntryWeigher implements EntryWeigher<Object, Object> {
```

Comment:
I think we should remove all unused ones here; being sub-classes they will generate class for each entry, and I have tried to keep `jackson-core` size to minimum.
I am bit worried about this PR adding lots of new classes for somewhat limited gain. So at least maybe we can trim features that are not needed -- I realize that keeping a forked version close to the original has benefits but maybe just commenting out these enum types would work.

Reply:
lol. My bad, this is for `jackson-databind`, so not quite that small set. Still ideally keeping things compact but bit less significant.

Reply:
I have removed some unused Weighers